src/Marten/Events/AsyncProjectionTestingExtensions.cs

```diff
@@ -26,7 +26,7 @@ public static async Task WaitForNonStaleProjectionDataAsync(this IDocumentStore
         else
         {
             var tasks = databases
-                .Select(db => Task.Factory.StartNew(() => db.WaitForNonStaleProjectionDataAsync(timeout))).ToArray();
+                .Select(db => db.WaitForNonStaleProjectionDataAsync(timeout)).ToArray();
```

Comment:
I'd drop the `ToArray` here*.

Currently `Task.WhenAll` will also create an array when an `IEnumerable<Task>` is given, but maybe in future the .NET implementation comes up with a solution that avoids that allocation, than Marten can benefit from this for free.

\* I see that this was there before

Reply:
Thanks, wasn't aware of this.
Remove it :)